components/builder-web/app/package/package-detail/package-detail.component.ts

```diff
@@ -138,8 +138,16 @@ export class PackageDetailComponent {
   }
 
   promotable(pkg) {
-    return this.memberOfOrigin &&
-    pkg.channels.length < this.store.getState().origins.current.channels.length;
+    const originChannels = this.store.getState().origins.current.channels;
+    let originChannelCount = originChannels.length;
+
+    originChannels.map((originChannel) => {
+      if (pkg.channels.indexOf(originChannel.name) !== -1) {
```

Comment:
I think you can do this like:
const hasMissingChannel = originChannels.some(originChannel =>
        pkg.channels.indexOf(originChannel.name) === -1
    );

    return this.memberOfOrigin && hasMissingChannel;

Reply:
[89b2f29](https://github.com/habitat-sh/builder/pull/1832/commits/89b2f2997f9edf197904f09c09f03c15da84a8de) resolved